extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/sql/MSQTaskSqlEngine.java

```diff
@@ -125,17 +125,20 @@ public boolean featureAvailable(EngineFeature feature, PlannerContext plannerCon
       case SCAN_NEEDS_SIGNATURE:
         return true;
       default:
-        throw new IAE("Unrecognized feature: %s", feature);
+        throw DruidException
+            .forPersona(DruidException.Persona.DEVELOPER)
+            .ofCategory(DruidException.Category.DEFENSIVE)
+            .build("Unrecognized feature: %s", feature);
```

Comment:
This doesn't seem defensive either?  This message seems intended for the admin or the operator perhaps?  That or it could be end user, do we document these "features" or are they intended to be hidden?

Reply:
These features are intended to be hidden from the user/admin/operator and are an implementation detail of Druid's planner. 
They are the properties of whatever the `SqlEngine` can or cannot support, which allows the different engines (Native and MSQ) to co-exist, and instruct the planner to plan a query in a specific way. For example, the MSQ engine doesn't support TopN queries, so by not supporting this feature, we instruct the planner to plan those as GroupBy queries.

As such, the only way we would ever encounter this is if we add a new `EngineFeature`, and we don't mention it's support in the `MSQTaskSqlEngine` or `NativeSqlEngine`, which is a developer's oversight and should have been flagged in the first place. Therefore it is a defensive exception.

Reply:
Adding the relevant portions from the above comment into this exception message, since these are developer-facing.